Trusted Origins API doc changes for iframe embedding feature (#3272)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/trusted-origins/index.md

The Okta Trusted Origins API provides operations to manage Trusted Origins and sources.

When external URLs are requested during sign-in, sign-out, or recovery operations, Okta checks those URLs against the allowed list of Trusted Origins. Trusted Origins also enable browser-based applications to access Okta APIs from JavaScript (CORS). If the origins aren't specified, the related operation (redirect or Okta API access) isn't permitted. You can also configure Trusted Origins to allow iFrame embedding of Okta resources, such as Okta sign-in pages and the Okta End-User Dashboard, within that origin.

## Trusted Origins API operations

### Create Trusted Origin

<ApiOperation method="post" url="/api/v1/trustedOrigins" />

Creates a new Trusted Origin

#### Valid request example

> ```json

Reviewer: ```suggestion
```bash
```

#### Invalid request example

> ```json

Reviewer: ```suggestion
```bash
```

}
```

#### Valid request example with iFrame embedding

<ApiLifecycle access="ea" />

Creates a new Trusted Origin for iFrame embedding of an Okta resource within that origin. In this example, the type of Okta resource is both the Okta End-User Dashboard and the Okta sign-in page.

```bash
curl -X POST \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "Authorization: SSWS ${api_token}" \
-d '{
  "name": "New Trusted Origin",
  "origin": "http://example.com",
  "scopes": [
    {
      "type": "CORS"
    },
    {
      "type": "REDIRECT"
    },
    {
      "type": "IFRAME_EMBED",
      "allowedOktaApps": [“OKTA_ENDUSER”]
    }
  ]
}' "https://${yourOktaDomain}/api/v1/trustedOrigins"
```

> Creates a new Trusted Origin for iFrame embedding of an Okta resource within that origin. In this example, the Okta resource is the Okta sign-in page.

Reviewer 2: Are the CORS and REDIRECT scopes required for this example?  If not, since the example pivots around iFrame embedding, maybe we could omit them.  Applies to examples below.

Author: i have updated the example

#### Request example

> ```json

Reviewer: ```suggestion
```bash
```

}
```

#### Request example with iFrame embedding

<ApiLifecycle access="ea" />

```bash
curl -X PUT \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "Authorization: SSWS ${api_token}" \
-d '{
  "id": "tosue7JvguwJ7U6kz0g3",
  "name": "Updated Example Trusted Origin",
  "origin": "http://updated.example.com",
  "scopes": [

>     {

Reviewer 2: Again, are the CORS and REDIRECT scopes required in this request.

Author: I have updated the example here too but i have some example which shows the combination of various scope types in the request body.

| id             | Unique identifier for the Trusted Origin                    | String                                    | No (assigned)   | N/A             |
| name           | Unique name for the Trusted Origin                          | String                                    | Yes             | 255 (chars)     |
| origin         | Unique origin URL for the Trusted Origin                    | String                                    | Yes             | 255 (chars)     |
| scopes         | Array of Scope types that this Trusted Origin is used for  | Array of [Scope objects](#scope-object)   | Yes             | 3 (Scope types) |

#### Scope object

Each Scope object specifies the type of Scope that its Trusted Origin is used for.

| Field Name  | Description                                                    | Data Type                         | Required |
| :---------- | :------------------------------------------------------------- | :-------------------------------- | :------- |

> | type        | The Scope type, which can be either "CORS" or "REDIRECT" or "IFRAME_EMBED". "IFRAME_EMBED" scope object can also have "allowedOktaApps" field of type array that can be left empty to allow embedding of sign-in pages only in an iframe, this field can also have this value ["OKTA_ENDUSER"] to allow embedding of both sign-in as well as EndUser dashboard pages in an iframe.                    | String                            | Yes      |

Reviewer: ```suggestion
| type        | The scope type. Supported values: `CORS`, `REDIRECT`, or `IFRAME_EMBED`.  You can also leave the `allowedOktaApps` property empty for the `IFRAME_EMBED` Scope object to allow the embedding of only sign-in pages in an iframe. This field can also include [`OKTA_ENDUSER`] as a value for that property to allow the embedding of both sign-in pages and the Okta End-User Dashboard in an iFrame.                    | String                            | Yes      |
```
